src/test/java/de/tum/in/www1/artemis/authentication/UserJenkinsGitlabIntegrationTest.java

```diff
@@ -276,6 +288,39 @@ public void createUserWithGroups() throws Exception {
         userTestService.createUserWithGroups();
     }
 
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void createUserWithGroupsAlreadyExistsInGitlab() throws Exception {
+        Course course = database.addEmptyCourse();
+        ProgrammingExercise programmingExericse = database.addProgrammingExerciseToCourse(course, false);
+
+        User newUser = userTestService.student;
+        newUser.setId(null);
+        newUser.setLogin("batman");
+        newUser.setEmail("foobar@tum.com");
+        newUser.setGroups(Set.of("tutor", "instructor"));
+
+        gitlabRequestMockProvider.mockAddUserToGroupsUserExists(newUser, programmingExericse.getProjectKey());
+        jenkinsRequestMockProvider.mockCreateUser(newUser, false, false, false);
+        request.post("/api/users", new ManagedUserVM(newUser), HttpStatus.CREATED);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void createUserWithGroupsAlreadyFailsInGitlab() throws Exception {
+        Course course = database.addEmptyCourse();
+        ProgrammingExercise programmingExericse = database.addProgrammingExerciseToCourse(course, false);
+
+        User newUser = userTestService.student;
+        newUser.setId(null);
+        newUser.setLogin("batman");
+        newUser.setEmail("foobar@tum.com");
```

Comment:
Maybe this part can be moved into a function since it gets repeated

Reply:
I don't really think it's necessary since I'm just setting values and the code isn't really repeated that often